Merging purgatory into osm_app branch

nshoemaker opened the pull request:
Because I cannot commit to the g3m repo myself, I am making a pull request so @boyP can pull my changes into our branch.

All I have done so far is merge the changes from purgatory into the osm_app branch. Also checked that the project builds correctly.

Commit:
Error in Jsonparser android

Commit:
...

Commit:
..

Commit:
...

Commit:
With texture

Commit:
...

Commit:
...

Commit:
...

Commit:
...

Commit:
...

Commit:
...

Commit:
...

Commit:
...

Commit:
...

Commit:
...

Commit:
...

Commit:
...

Commit:
...

Commit:
...

Commit:
...

Commit:
....

Commit:
...

Commit:
...

Commit:
...

Commit:
-

Commit:
Generated

Commit:
...

Commit:
...

Commit:
Generated

Commit:
...

Commit:
Generated

Commit:
...

Commit:
Generated

Commit:
added NonOverlapping demo to G3MApp - added removeAllMarks method to NonOverlappingMarksRenderer

Commit:
added a warning for JM

Commit:
-

Commit:
-

Commit:
...

Commit:
Trying Agustin approach

DiegoGomezDeck committed:
Update README.md

DiegoGomezDeck committed:
Update README.md

Commit:
converting

Commit:
Merge branch 'no-marks' of github.com:glob3mobile/g3m into no-marks

Commit:
...

Commit:
Generated

Commit:
clenaups

Commit:
cleanups

Commit:
Generated

Commit:
Generated

Commit:
...

Commit:
testing in Android

Commit:
testing in Android

Commit:
removed camera public access to mutable state

Commit:
Generated

Commit:
-

Commit:
-

Commit:
Generated

Commit:
Merge branch 'no-marks' into purgatory

Commit:
-

Commit:
-

Commit:
-

Commit:
first version of LayoutImageBuilder

Commit:
Generated

Commit:
first version of LayoutImageBuilder - convertion fixes

Commit:
Generated

Commit:
first version of LayoutImageBuilder - convertion fixes

Commit:
first version of LayoutImageBuilder - convertion fixes

Commit:
Generated

Commit:
-

Commit:
-

Commit:
-

Commit:
-

Commit:
Generated

Commit:
-

Commit:
Generated

Commit:
-

Commit:
-

Commit:
Generated

Commit:
-

Commit:
-

Commit:
Generated

Commit:
new StaticImageBuilder

Commit:
Generated

Commit:
updated .gitignore

Commit:
updated .gitignore

Commit:
added NonOverlappingMarksVisibilityListener - new clear() method in IStringBuilder

Commit:
Generated

Commit:
new methods NonOverlappingMark::setID()/getID()

Commit:
implemented clear() methods in StringBuilders

Commit:
Generated

Commit:
-

Commit:
Generated

Commit:
-

Commit:
fixed GenericQuadTree_Node::splitNode

Commit:
bugs fixed on Canvas_WebGL

Commit:
Generated

Commit:
-

Commit:
-

Commit:
-

Commit:
-

Commit:
-

Commit:
Merge branch 'purgatory' of github.com:glob3mobile/g3m into purgatory

Commit:
-

Commit:
Merge branch 'purgatory' of github.com:glob3mobile/g3m into purgatory

Commit:
-

Commit:
Generated

Commit:
-

Commit:
-

Commit:
Generated

Commit:
-

Commit:
-

Commit:
Generated

Commit:
-

Commit:
-

Commit:
-

Commit:
Generated

Commit:
-

Commit:
Generated

Commit:
-

Commit:
Escaping Percent Url String

Commit:
-

Commit:
new methods in Planet to reduce garbage

Commit:
Generated

Commit:
-

Commit:
-

Commit:
Generated

Commit:
-

Commit:
-

Commit:
-

Commit:
-

Commit:
-

Commit:
Generated

Commit:
-

Commit:
-

Commit:
Generated

Commit:
-

Commit:
Generated

Commit:
-

Commit:
-

Commit:
-

Commit:
Generated

Commit:
Generated

Commit:
Merge branch 'purgatory' into no-marks

Commit:
removed 2 memory leaks

Commit:
Generated

Commit:
-

Commit:
Generated

Commit:
-

Commit:
Generated

Commit:
removing 64 bits warnings

Commit:
-

Commit:
Generated

Commit:
removed memory leaks

Commit:
Generated

Commit:
removed memory leaks

Commit:
Generated

Commit:
new demo 'Augmented Reality'

Commit:
-

Commit:
-

Commit:
Generated

Commit:
Generated

Commit:
-

Commit:
-

Commit:
-

Commit:
-

Commit:
-

Commit:
-

Commit:
-

Commit:
Generated

Commit:
Generated

Commit:
-

Commit:
-

Commit:
Generated

Commit:
Generated

Commit:
-

Commit:
-

Commit:
-

Commit:
-

Commit:
Generated

Commit:
Generated

Commit:
updated non-overlapping demo

Commit:
updated non-overlapping demo

Commit:
updated non-overlapping demo

Commit:
updated non-overlapping demo

Commit:
-

Commit:
-

Commit:
Generated

Commit:
Generated

Commit:
added GA tracking

Commit:
added GA tracking

Commit:
-

Commit:
-

Commit:
-

Commit:
-

Commit:
Generated

Commit:
Generated

Commit:
DisplayInfo

Commit:
several minor changes (and fixes) for Info displaying

Commit:
several minor changes (and fixes) for Info displaying

Commit:
Generated

Commit:
Generated

Commit:
-

Commit:
-

Commit:
Generated

Commit:
Generated

Commit:
-

Commit:
-

Commit:
Merge branch 'purgatory' into ar

Conflicts:
	iOS/G3MiOSSDK/Commons/Rendererers/PlanetRenderer.cpp

Commit:
-

Commit:
Generated

Commit:
-

Commit:
bug fixed in G3MApp.xcodeproj

Commit:
Merging fixed purgatory into osm_app. Everything builds

Commit:
Finding bounding box for each building

Commit:
Finding bounding box for each building

Commit:
Merge branch 'osm_app' of https://github.com/nshoemaker/g3m into osm_app

nshoemaker committed:
Adding comment about bounding box in order to check if I am now the one committing

nshoemaker committed:
Started creating BoxShape for buildings. Not being placed correctly. Still need to debug a bit.

nshoemaker commented:
I have also started creating BoxShapes for each building. Currently testing on the first 10 buildings of each tile. Currently, the BoxShapes seem to be located at places other than where I would want them to be. Not sure if I am feeding position in wrong, or if things are growing upside down, through the earth, and back out the other side. Still need to debug.

nshoemaker commented on iOS/G3MApp/G3MApp/G3MOSMBuildingsDemoScene.cpp:
```diff
@@ -18,9 +18,16 @@
 #include <G3MiOSSDK/JSONArray.hpp>
 #include <G3MiOSSDK/JSONNumber.hpp>
 #include <G3MiOSSDK/Mark.hpp>
+#include <G3MiOSSDK/Box.hpp>
```
Get rid of this we are no longer using the Box class.

DiegoGomezDeck commented:
> Currently, the BoxShapes seem to be located at places other than where I would want them to be.

Take in mind the "zero" of the Box's coordinate-system is its center.

nshoemaker commented:
Thanks, @DiegoGomezDeck! We realized this and moved our position coordinates to the center of the box. 

Originally, we were actually running into undefined behavior due to pointer issues with the position coordinates we gave the BoxShape. 

We just fixed both problems though and so far, so good.

boyP commented:
![ios simulator screen shot apr 11 2015 3 58 55 pm](https://cloud.githubusercontent.com/assets/6115468/7103037/c50b8b4c-e063-11e4-929f-3632eb263fc9.png)
